File: SportHub_server/SportHub.Core/Entities/Comment.cs
using SportHub.Core.Entities.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace SportHub.Core.Entities
{
    public class Comment : Entity
    {
        public string Text { get; set; }
        public DateTime DateCreated { get; set; }
        public bool IsUpdated { get; set; }

        public Guid UserId { get; set; }
        public User User { get; set; }

        public int PostId { get; set; }
        public Post Post { get; set; }

        public Comment(string text, DateTime dateCreated, bool isUpdated, Guid userId, int postId)
        {
            Text = text;
            DateCreated = dateCreated;
            IsUpdated = isUpdated;
            UserId = userId;
            PostId = postId;
        }
    }
}

File: SportHub_client/lib/entities/comment.dart
import 'package:SportHub_client/entities/user.dart';

class Comment {
  int id;
  String text;
  String dateCreated;
  bool isUpdated;
  String userId;
  int postId;
  User user;

  Comment(
      {this.id,
      this.text,
      this.dateCreated,
      this.isUpdated,
      this.userId,
      this.postId,
      this.user});

  Comment.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    text = json['text'];
    dateCreated = json['dateCreated'];
    isUpdated = json['isUpdated'];
    userId = json['userId'];
    postId = json['postId'];
    user = json['user'] != null ? new User.fromJson(json['user']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['text'] = this.text;
    data['dateCreated'] = this.dateCreated;
    data['isUpdated'] = this.isUpdated;
    data['userId'] = this.userId;
    data['postId'] = this.postId;
    if (this.user != null) {
      data['user'] = this.user.toJson();
    }
    return data;
  }
}
